### ISSUE TITLE
no-cycle: TypeError aborts the lint run when the linted path is not on disk (stdin, lintText)

### ISSUE THREAD

**Issue by theRizwan** [benmosher/eslint-plugin-import#3286]
`import/no-cycle` throws a `TypeError` and aborts the entire lint run whenever the linted path does
not exist on disk — which is the normal case for an unsaved buffer, generated code, or
`ESLint#lintText`.

```
$ cat src/foo.js | npx eslint --stdin --stdin-filename=src/not-yet-written.js

Oops! Something went wrong! :(

ESLint: 9.39.5

TypeError: Cannot read properties of null (reading '/…/src/not-yet-written.js')
Occurred while linting /…/src/not-yet-written.js:1
Rule: "import/no-cycle"
```

Exit code 2, and no results are reported for any file — not just the one that triggered it.

Reproduced on a clean install of `eslint-plugin-import@2.32.0` with `eslint@9.39.5`, and present at
`main` (`1a39fb8`).

### Cause

`src/rules/no-cycle.js:73`

```js
const scc = options.disableScc ? {} : StronglyConnectedComponentsBuilder.get(myPath, context);
```

`StronglyConnectedComponentsBuilder.get` returns `null` when the path does not resolve
(`src/scc.js:16` — `if (path == null) { return null; }`). Line 116 then does `scc[myPath]`.

Two conditions are needed together, which is why it is easy to miss: the linted path must not
resolve, **and** at least one import in the file must resolve — otherwise `checkSourceValue` returns
early at the `imported == null` check before reaching line 116.

The `<text>` guard on line 63 shows non-file input was anticipated. That sentinel is the eslintrc
one; under flat config the buffer keeps whatever `--stdin-filename` supplies, so the guard no longer
covers this.

### Suggested fix

Defaulting the map to `{}` restores exactly the `disableScc` semantics: every lookup is `undefined`,
so each pair compares equal and everything is traversed. The SCC map is only an optimisation for
skipping unreachable imports, so this is safe rather than merely defensive.

```diff
-    const scc = options.disableScc ? {} : StronglyConnectedComponentsBuilder.get(myPath, context);
+    const scc = options.disableScc ? {} : StronglyConnectedComponentsBuilder.get(myPath, context) || {};
```

Happy to open a PR — I have one ready with a regression test. Confirmed the test fails without the
change (`1 failing`, same `TypeError`) and passes with it; full rule suite 2877 passing, lint clean.

Possibly related: #3221 reports the same crash site with a resolver-specific trigger, attributed
there to `eslint-import-resolver-exports`. This reproduction uses the default resolver and no extra
plugins, so the root cause may be shared.

Environment: `eslint-plugin-import@2.32.0`, `eslint@9.39.5`, Node 24.16.0, macOS.
